### PR TITLE
CI-100: Emotion Intensity

### PR DESCRIPTION
**Pull request by LyubovP**
[CI-100](https://cloverpop-internship.atlassian.net/browse/CI-100)

Link to Epic Requirements https://docs.google.com/document/d/1gtgo-KAjz6IBQAMZoSu4__hi3zsjuInPRAHarXCDmc0/edit

[CI-100]: https://cloverpop-internship.atlassian.net/browse/CI-100?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/javascript/components/Pages/EmotionIntensity.js [hunk 1]
```diff
@@ -29,23 +70,58 @@ const EmotionIntensity = ({data, setData, saveDataToDb, steps, service}) => {
...
-    <span className="under-convert">5</span>
-  </div>
+  const generateStyles = (value, selected, category) => {
+    let backgroundColor, borderColor;
+    switch(category) {
+      case "negative":
+        backgroundColor = 
+          value === 5 ? '#F18C59' : value === 4 ? '#F09D74' :
+          value === 3 ? '#F5C1A6' : value === 2 ? '#F7CDB8' :
+          value === 1 ? '#FADFD1' : 'transparent';
```
Comment (rogergraves) [inline app/javascript/components/Pages/EmotionIntensity.js:80]: Can you please explain what is going on here?

Comment (LyubovP) [reply app/javascript/components/Pages/EmotionIntensity.js:80]: @rogergraves, yes of course I can explain. generateStyles which takes three arguments: value, selected, and category. The purpose of this function is to generate a style object based on the input arguments. generateStyles function returns an object with three properties: backgroundColor, borderRadius, and border.
The function uses a switch statement to determine which category the style should be generated for. The three possible categories are "negative", "neutral", and "positive".
For each category, the function sets a backgroundColor and a borderColor based on the value argument 1 to 5, and word category.

File: app/javascript/components/Pages/EmotionIntensity.js [hunk 2]
```diff
@@ -1,51 +1,162 @@
...
+        {[1, 2, 3, 4, 5].map((value) => (
+          <label key={value} className="rating-label" style={generateStyles(value, rating === value, category)}>
+            <input
+              type="radio"
+              name="rating"
+              value={value}
+              checked={rating === value}
+              onChange={() => handleRatingClick(value)}
+            />
+            {value}{value !== 1}
```
Comment (SokalMet) [inline app/javascript/components/Pages/EmotionIntensity.js:33]: are you sure you need **{value !== 1}** here?

**Review by SokalMet — CHANGES_REQUESTED** [review]

File: app/javascript/components/Pages/EmotionIntensity.js (hunk 2 above)
Comment (LyubovP) [reply app/javascript/components/Pages/EmotionIntensity.js:33]: yes, expression {value !== 1}  is not necessary

File: app/assets/stylesheets/gif.scss [hunk 3]
```diff
@@ -124,4 +124,26 @@
...
+  top: 0;
+  right: 0;
+  width: 90px;
+  height: 10px;
+}
+
+.gif {
+  display: inline-block;
+  text-align: right;
 }
```
[the remark this answers is not in the document]
Comment (LyubovP) [reply app/assets/stylesheets/gif.scss:145]: I tried using text-right, but the styles are not being applied properly to the text-right class. I added style={{ textAlign: 'right' }} where I needed to align the "PoweredBy" image to the right edge.

**Review by SBorozenets — CHANGES_REQUESTED** [review]

**Review by rogergraves — APPROVED** [review]

File: app/assets/stylesheets/gif.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (LyubovP) [reply app/assets/stylesheets/gif.scss]: I added "test-end" where I needed to align the "PoweredBy" image to the right edge.

**Review by SokalMet — APPROVED** [review]

**Pull request merged**